### PR TITLE
[mtl] debug output for descriptor allocation and frame wait times

### PR DESCRIPTION
**Pull request by kvark**
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends:

### CODE REVIEW

File: src/backend/metal/src/device.rs [hunk 1]
```diff
@@ -125,6 +125,7 @@ pub struct Device {
     pub(crate) shared: Arc<Shared>,
     pub(crate) private_caps: PrivateCapabilities,
     memory_types: [hal::MemoryType; 4],
+    debug_min_descriptor_count: usize,
```
Comment (grovesNL) [inline src/backend/metal/src/device.rs]: Was this left in this PR intentionally?

File: src/backend/metal/src/command.rs [hunk 2]
```diff
@@ -1373,6 +1393,18 @@ impl RawCommandQueue<Backend> for CommandQueue {
 
         command_buffer.commit();
 
+        if let Some(ref mut wait) = self.frame_wait {
+            wait.frame += 1;
+            if wait.frame >= FRAME_WAIT_REPORT_WINDOW {
+                let time = wait.duration / wait.frame as u32;
+                println!("Frame wait time={}ms count={}",
```
Comment (grovesNL) [inline src/backend/metal/src/command.rs]: The `println` could be undesirable for some applications if they're ok with this

**Comment by kvark** [comment]
The printlns are currently blocked by a compile time constant, so shouldn't be a problem for now.

Bors r=grovesNL

File: src/backend/metal/src/device.rs (hunk 1 above)
Comment (kvark) [reply src/backend/metal/src/device.rs]: oh yes, this is an unfortunate workaround we have to get until Dota is fixed

**Pull request merged**